propka/group.py

```diff
@@ -344,1005 +330,1090 @@ def clone(self):
         else:
-            return self.interaction_atoms_for_acids #default is acid interaction atoms - cf. 3.0
+            # default is acid interaction atoms - cf. 3.0
+            return self.interaction_atoms_for_acids
 
     def set_center(self, atoms):
+        """Set center of group based on atoms.
+
+        Args:
+            atoms:  list of atoms
+        """
         if not atoms:
             raise ValueError("At least one atom must be specified")
-
         # reset center
-        self.x = 0.0; self.y = 0.0; self.z = 0.0
-
-        # find the average positon of atoms
+        self.x = 0.0
+        self.y = 0.0
+        self.z = 0.0
+        # find the average position of atoms
         for atom in atoms:
-            self.x += atom.x; self.y += atom.y; self.z += atom.z
-
+            self.x += atom.x
+            self.y += atom.y
+            self.z += atom.z
         self.x /= float(len(atoms))
         self.y /= float(len(atoms))
         self.z /= float(len(atoms))
 
-        return
+    def get_determinant_string(self, remove_penalised_group=False):
+        """Create a string to identify this determinant.
 
-    def getDeterminantString(self, remove_penalised_group=False):
+        Args:
+            remove_penalised_group:  Boolean flag to remove penalized groups
+        Returns:
+            string
+        """
         if self.coupled_titrating_group and remove_penalised_group:
             return ''
-
         number_of_sidechain = len(self.determinants['sidechain'])
-        number_of_backbone  = len(self.determinants['backbone'])
-        number_of_coulomb   = len(self.determinants['coulomb'])
-
-        number_of_lines     = max(1, number_of_sidechain, number_of_backbone, number_of_coulomb)
-        str  = ""
+        number_of_backbone = len(self.determinants['backbone'])
+        number_of_coulomb = len(self.determinants['coulomb'])
+        number_of_lines = max(1, number_of_sidechain, number_of_backbone,
+                              number_of_coulomb)
+        str_ = ""
         for line_number in range(number_of_lines):
-            str += "%s" % (self.label)
+            str_ += "{0:s}".format(self.label)
             if line_number == 0:
-                str += " %6.2lf" %(self.pka_value)
-                if len(self.non_covalently_coupled_groups)>0:
-                    str+='*'
+                str_ += " {0:6.2f}".format(self.pka_value)
+                if len(self.non_covalently_coupled_groups) > 0:
+                    str_ += '*'
                 else:
-                    str+=' '
-
-                # if self.atom.cysteine_bridge:
-                #     str += " BONDED "
-                # else:
-                str += " %4d%2s " % (int(100.0*self.buried), "%")
-
-                str += " %6.2lf %4d" % (self.Emass, self.Nmass)
-                str += " %6.2lf %4d" % (self.Elocl, self.Nlocl)
+                    str_ += ' '
+                str_ += " {0:4d}{1:>2s} ".format(int(100.0*self.buried), "%%")
```

Comment:
this puts `%%` in the output

Reply:
Addressed by https://github.com/Electrostatics/propka-3.1/commit/d7147799d815f6f273b4814419cd8d5955ad4569